pkg/viewer/overview.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2023 The Bestchains Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package viewer
+
+import (
+	"fmt"
+
+	"github.com/bestchains/bc-explorer/pkg/models"
+	"github.com/go-pg/pg/v10"
+)
+
+const (
+	BlockAggregation       = "blocks"
+	TransactionAggregation = "transactions"
+)
+
+type SummaryResp struct {
+	BlockNumber uint64 `pg:"blockNumber" json:"blockNumber"`
+	TxCount     uint64 `pg:"txCount" json:"txCount"`
+}
+
+type BySegResp struct {
+	Start int64 `json:"start"`
+	End   int64 `json:"end"`
+	Count int64 `json:"count"`
+}
+
+type Overview interface {
+	// Summary returns block height, number of transactions, number of nodes, total number of contracts.
+	Summary() (SummaryResp, error)
+
+	// QueryBySeg query the total number of transactions or blocks for a number of time periods
+	// from, interval,number of time periods
```

Comment:
have you sync with @Jandiasnow  with theses arguments ?

Reply:
Synced.